lib/puppet/face/config.rb

```diff
@@ -106,6 +106,20 @@
     EOT
 
     when_invoked do |name, value, options|
+      if name == 'environment' && options[:section] == 'main'
+        Puppet.warning _(<<-EOM).chomp
+The environment should be set in either the `[user]`, `[agent]`, or `[master]`
+section. Variables set in the `[agent]` section are used when running
+`puppet agent` and do not require the defined environment directory to exist
+locally. Variables set in the `[user]` section are used when running various
```

Comment:
That ended up wordier than I'd like, but ok.

Reply:
I trimmed it down a bit. I just removed the information about puppet agent not requiring environments to exist locally.